examples/test_read_input.rs

```diff
@@ -0,0 +1,56 @@
+extern crate midir;
+
+use std::io::{stdin, stdout, Write};
+use std::error::Error;
+
+use midir::{MidiInput, Ignore};
+
+fn main() {
+    match run() {
+        Ok(_) => (),
+        Err(err) => println!("Error: {}", err.description())
+    }
+}
+
+fn run() -> Result<(), Box<Error>> {
+    let mut input = String::new();
+    
+    let mut midi_in = MidiInput::new("midir reading input")?;
+    midi_in.ignore(Ignore::None);
+    
+    // Get an input port (read from console if multiple are available)
+    let in_port = match midi_in.port_count() {
+        0 => return Err("no input port found".into()),
+        1 => {
+            println!("Choosing the only available input port: {}", midi_in.port_name(0).unwrap());
+            0
+        },
+        _ => {
+            println!("\nAvailable input ports:");
+            for i in 0..midi_in.port_count() {
+                println!("{}: {}", i, midi_in.port_name(i).unwrap());
+            }
+            print!("Please select input port: ");
+            stdout().flush()?;
+            let mut input = String::new();
+            stdin().read_line(&mut input)?;
+            input.trim().parse()?
+        }
+    };
+    
+    println!("\nOpening connections");
```

Comment:
Just a nit: because there's now only one connection, it should say "Opening connection" and the same below for "Closing connection".